.github/workflows/docker-build.yml

```diff
@@ -0,0 +1,172 @@
+
+      - name: Generate build matrices
+        id: set-matrix
+        run: |
+          # Get list of changed Dockerfiles
+          CHANGED_FILES=$(echo "${{ steps.changed-files.outputs.all_changed_files }}" | tr ' ' '\n')
+          
+          # Create matrices using jq
+          echo "base_matrix=$(echo "$CHANGED_FILES" | grep "^node-base/" | sed 's|/Dockerfile$||' | jq -Rsc 'split("\n")[:-1] | {include: map({image_path: .})}')" >> $GITHUB_OUTPUT
+          echo "clients_matrix=$(echo "$CHANGED_FILES" | grep "^clients/" | sed 's|/Dockerfile$||' | jq -Rsc 'split("\n")[:-1] | {include: map({image_path: .})}')" >> $GITHUB_OUTPUT
+          echo "protocols_matrix=$(echo "$CHANGED_FILES" | grep "^ethereum/" | sed 's|/Dockerfile$||' | jq -Rsc 'split("\n")[:-1] | {include: map({image_path: .})}')" >> $GITHUB_OUTPUT
+
+  build-base:
+    needs: detect-changes
+    if: ${{ fromJson(needs.detect-changes.outputs.base_matrix).include[0] }}
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJson(needs.detect-changes.outputs.base_matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Generate version
+        id: version
+        run: |
+          SHA=$(git rev-parse --short HEAD)
+          echo "image_name=node-base" >> $GITHUB_OUTPUT
+          echo "image_tag=${SHA}" >> $GITHUB_OUTPUT
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Build and push base image
+        uses: docker/build-push-action@v5
+        with:
+          context: ./${{ matrix.image_path }}
+          push: true
+          build-args: |
+            GRAFANA_LOKI_API_KEY=${{ secrets.GRAFANA_LOKI_API_KEY }}
+            GRAFANA_PROM_API_KEY=${{ secrets.GRAFANA_PROM_API_KEY }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+          tags: ghcr.io/blockjoy/${{ steps.version.outputs.image_name }}:${{ steps.version.outputs.image_tag }}
+
+  build-clients:
+    needs: [detect-changes, build-base]
+    if: |
+      always() &&
+      needs.build-base.result != 'failure' &&
+      fromJson(needs.detect-changes.outputs.clients_matrix).include[0]
+    runs-on: ubuntu-latest
+    strategy:
+      matrix: ${{ fromJson(needs.detect-changes.outputs.clients_matrix) }}
+    steps:
+      - uses: actions/checkout@v4
+      - name: Log in to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Generate version
+        id: version
+        run: |
+          SHA=$(git rev-parse --short HEAD)
+          CLIENT_NAME=$(basename ${{ matrix.image_path }})
+          IMAGE_NAME="blockjoy-${CLIENT_NAME}"
```

Comment:
The namespace is already /blockjoy here because that's the organization's name, I added blockjoy- again to distinguish it from a potential vanilla erigon image

Reply:
I'm not a fan of `blockjoy/blockjoy-${CLEINT_NAME}`.  I like the intent though. Are customers going to be pushing their images to the blockjoy namespace?  If not then the extra is redundant.  Maybe we use a different namespace for community created images.  Otherwise there's nothing stopping someone (other than us) from creating `blockjoy/blockjoy-malicious-image`.  Or overwriting `blockjoy/blockjoy-reth` with a malicious or bad image.